btf/btf.go

```diff
@@ -228,8 +228,19 @@ func loadSpecFromELF(file *internal.SafeELFFile) (*Spec, error) {
 	return spec, nil
 }
 
-func loadRawSpec(btf io.ReaderAt, bo binary.ByteOrder,
-	baseTypes types, baseStrings *stringTable) (*Spec, error) {
+func loadRawSpec(btf io.ReaderAt, bo binary.ByteOrder, base *Spec) (*Spec, error) {
+	var (
+		baseStrings *stringTable
+		baseTypes   []Type
+		firstTypeID TypeID
+	)
+
+	if base != nil {
+		baseStrings = base.strings
+		baseTypes = base.types
+		firstTypeID = base.lastTypeID + 1
```

Comment:
Shouldn't this be:
```suggestion
		firstTypeID = TypeID(len(baseTypes))
```
Or am I missing something?

Reply:
Drive by "possible bugfix" by me, sorry. I should've split this out into a separate commit. `lastTypeID` is more reliable, since `len(baseTypes)` assumes that types start at 0. That isn't correct for split BTF. Now, we don't support split BTF base, but I figured it's worth to fix this before it becomes a bug.

Reply:
Made a separate commit.

Reply:
It's all broken :cry: